src/linker/Linker/DependencyInfo.cs

```diff
@@ -131,6 +131,11 @@ public enum DependencyKind
 		DisablePrivateReflectionRequirement = 79, // null -> DisablePrivateReflectionAttribute type/methods (note that no specific source is reported)
 		DynamicInterfaceCastableImplementation = 80, // type -> type is marked with IDynamicInterfaceCastableImplementationAttribute and implements the provided interface
 		AlreadyMarked = 82, // null -> member that has already been marked for a particular reason (used to propagate reasons internally, not reported)
+
+		DataContractSerialized = 83, // entry type for DataContract serialization
+		XmlSerialized = 84, // entry type for XML serialization
+		RecursiveType = 85, // recursive type kept dup to serialization handling
```

Comment:
Nit: I would probably rename to something like SerializedRecursiveType - just to make it clear it's for serialization only